course4/SinglyLinkedList/sllist.hh

```diff
@@ -0,0 +1,112 @@
+#pragma once
+#include <memory>
+#include <iostream>
+
+namespace course {
+
+template<class T>
+struct Node
+{
+    T mData;
+    std::unique_ptr<Node> mNext;
+};
+
+template<class T>
+class iterator: public std::iterator<std::forward_iterator_tag, T>
+{
+private:
+    Node<T>* p;
+public:
+    iterator(Node<T>* p) : p(p) {}
+    iterator& operator=(iterator other)
+    {
+        std::swap(p, other.p);
+        return *this;
+    }
+    const T& operator*() const { return p->mData; }
+    bool operator==(const iterator& other) { return p == other.p; }
+    bool operator!=(const iterator& other)
+    {
+        //std::cout<<p<<" != "<<other.p<<std::endl;
```

Comment:
Remove all the comment-out  code. You can always retrieve it from revision history, if needed.